src/librustc_mir/transform/inline.rs

```diff
@@ -532,8 +528,58 @@ impl<'a, 'tcx> Inliner<'a, 'tcx> {
                 let return_block = destination.1;
 
                 // Copy the arguments if needed.
-                let args : Vec<_> = {
+                let args : Vec<_> = if is_box_free {
```

Comment:
Oh, yuck! Shouldn't this be done at HIR→MIR lowering time?

This is fine for now, but I'd factor it out into a separate function, with a big ol' FIXME on top of it :)